### PR TITLE
feat: upgrade to palm

### PR DESCRIPTION
**Pull request by irtazaakram**

### CODE REVIEW

**Review by regisb — COMMENTED** [review]
Thanks for the PR @irtazaakram! Jhony, the maintainer of this plugin, just opened a similar PR here: https://github.com/overhangio/tutor-xqueue/pull/16
You should definitely add your changes on top on Jhony's PR. Would you mind rebasing your branch on top of the "palm" branch and modify this PR to use the "palm" branch as base?

**Comment by irtazaakram** [comment]
@regisb Done 👍🏻

**Review by regisb — APPROVED** [review]
LGTM! @jfavellar90 what do you think? This PR has a few not-so-important formatting changes, but also includes cool improvements, for instance in the Dockerfile.

File: tutorxqueue/plugin.py [hunk 1]
```diff
@@ -256,22 +258,36 @@ def request(self, endpoint, method="GET", data=None, params=None):
...
+# Xqueue Public Host
+########################################
+
+
+@tutor_hooks.Filters.APP_PUBLIC_HOSTS.add()
+def _discovery_public_hosts(
+    hosts: list[str], context_name: t.Literal["local", "dev"]
+) -> list[str]:
+    if context_name == "dev":
+        hosts += ["discovery.{{ LMS_HOST }}:8000"]
```
Comment (jfavellar90) [inline tutorxqueue/plugin.py]:
```suggestion
        hosts += ["xqueue.{{ LMS_HOST }}:8000"]
```
@irtazaakram the same fix should be applied 2 lines below

**Review by jfavellar90 — CHANGES_REQUESTED** [review]

File: tutorxqueue/plugin.py (hunk 1 above)
Comment (irtazaakram) [reply tutorxqueue/plugin.py]: @jfavellar90 Done 👍🏻

**Comment by jfavellar90** [comment]
@irtazaakram thanks for your contribution. Could you please rebase again on top of the "palm" branch? This is because @regisb added a new commit in this branch

**Comment by irtazaakram** [comment]
@jfavellar90 Done 👍🏻

**Pull request merged**